test: Adds tests and storybook to RefreshLabel

michael-s-molina opened the pull request:
### SUMMARY
- Coverts `RefreshLabel` to functional component with typescript
- Adds tests and storybook to `RefreshLabel`
- Changes `RefreshLabel` to use the new `Icon` component
- Removes `less` files

### TEST PLAN
1 - Execute `RefreshLabel` tests
2 - All tests should pass

@rusackas 

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

michael-s-molina committed:
test: Adds tests and storybook to RefreshLabel

rusackas commented on superset-frontend/src/components/RefreshLabel/index.tsx:
```diff
@@ -16,32 +16,38 @@
+        role="button"
+        onClick={onClick}
+        name="refresh"
+        css={{
```
Wait... we're not importing `css` from `@superset-ui/core` here... is this css prop going unheard or something?

Also, if we're going to use the `css` prop, I kind of like the pattern of:
```
css={theme => css`
  cursor:  pointer;
  color: ${theme.colors.grayscale.base};
  &:hover {
    color: ${theme.colors.grayscale.base};
  }
`}
```

this just seems a little more readable to me, and uses real css (e.g. `border-color`) rather than the JS flavor (e.g. `borderColor`) and is copy/pastable to and from `styled` implementations.

michael-s-molina replied:
We don't need to import [`css`](https://emotion.sh/docs/css-prop) when using object styles, only when we use string styles. Babel will automatically convert this property to a `className`.

I'll disagree with you about string style vs object style. What really makes the difference for me when I chose object style is typescript checking and autocompletion. You can't do that when using string styles and that is more error-prone. Also, for me at least, is more natural coming from React and React Native style pattern.

rusackas replied:
> About CSS-y approach I understand that it doesn't require switching the brain from CSS to JS styles but I still think that type checking is a clear winner. Let me know if you agree. Otherwise, I can revisit. Thank you for your review.

The ability to more easily port from `css` to `styled` or back was the bigger selling point, but type checking probably wins over that. Seems OK to me. 

As for autocomplete, I have an IDE plugin (`ts-styled-plugin` if memory serves) that takes care of intellisense/autocomplete with the styled/string implementation, which is helpful :)

rusackas reviewed (commented):
Pointed out one potential risk, and saw two style nits:

- How we use the `css` prop: I kind of prefer the CSS-y approach, since it's consistent with the `styled` approach and doesn't require switching the brain from CSS to JS styles.
- a few `3px` measurements... I hate to be _that guy_ but should these be a `gridUnit`?

michael-s-molina commented:
@rusackas I changed to use `gridUnit` instead of `3px` and also replaced `useTheme` with typed `theme: SupersetTheme` prop 😉. 

About CSS-y approach I understand that it doesn't require switching the brain from CSS to JS styles but I still think that type checking is a clear winner. Let me know if you agree. Otherwise, I can revisit. Thank you for your review.

rusackas commented on superset-frontend/src/components/RefreshLabel/RefreshLabel.stories.tsx:
```diff
@@ -16,12 +16,29 @@
  * under the License.
  */
-@import '../../stylesheets/less/variables.less';
+import React from 'react';
```
Weird that git thought this was a file rename, but... well... ok.
